python/cuml/dask/ensemble/randomforestcommon.py

```diff
@@ -0,0 +1,173 @@
+                key="%s-%s" % (key, n),
+                workers=[worker],
+            )
+            for n, worker in enumerate(self.workers)
+        }
+
+        rfs_wait = list()
+        for r in self.rfs.values():
+            rfs_wait.append(r)
+
+        wait(rfs_wait)
+        raise_exception_from_futures(rfs_wait)
+
+    def _fit(self, model, dataset, convert_dtype):
+        data = DistributedDataHandler.create(dataset, client=self.client)
+        self.datatype = data.datatype
+        futures = list()
+        for idx, (worker, worker_data) in \
+                enumerate(data.worker_to_parts.items()):
+            futures.append(
+                self.client.submit(
+                    _func_fit,
+                    model[worker],
+                    worker_data,
+                    convert_dtype,
+                    workers=[worker],
+                    pure=False)
+            )
+        wait(futures)
+        raise_exception_from_futures(futures)
+        return self
+
+    def _concat_treelite_models(self):
+        """
+        Convert the cuML Random Forest model present in different workers to
+        the treelite format and then concatenate the different treelite models
+        to create a single model. The concatenated model is then converted to
+        bytes format.
+        """
+        mod_bytes = []
+        for w in self.workers:
+            mod_bytes.append(self.rfs[w].result().model_pbuf_bytes)
+        last_worker = w
+        all_tl_mod_handles = []
+        model = self.rfs[last_worker].result()
+        for n in range(len(self.workers)):
+            all_tl_mod_handles.append(model._tl_model_handles(mod_bytes[n]))
+        concat_model_handle = model._concatenate_treelite_handle(
+            treelite_handle=all_tl_mod_handles)
+
+        model._concatenate_model_bytes(concat_model_handle)
+        return model
+
+    def _predict_using_fil(self, X, delayed, **kwargs):
+        data = DistributedDataHandler.create(X, client=self.client)
+        self.datatype = data.datatype
+        return self._predict(X, delayed=delayed, **kwargs)
+
+    def _get_params(self, deep):
+        model_params = list()
+        for n, worker in enumerate(self.workers):
+            model_params.append(
+                self.client.submit(
+                    _func_get_params,
+                    self.rfs[worker],
+                    deep,
+                    workers=[worker]
+                )
+            )
+        wait(model_params)
+        raise_exception_from_futures(model_params)
+        params_of_each_model = list()
+        for i in range(len(model_params)):
+            params_of_each_model.append(model_params[i].result())
+        return params_of_each_model
+
+    def _set_params(self, worker_numb, **params):
+        model_params = list()
+        if worker_numb is None:
+            worker_numb = [posi for posi in range(len(self.workers))]
+        workers = [worker for worker in self.workers]

```

Comment:
This is equivalent to `workers = self.workers`?

Reply:
the workers here is reading the information from self.workers variable and saving the values in a list. This is required as self.workers is of type dict.keys(). 
This will not be required if we are going to remove the option of letting the user decide the workers for which the parameters should be set.

Reply:
Ok, I see this is converting to a list. I think then `workers = list(self.workers)` would be nice and clear.

Reply:
I thought we did not want to give the user the option of picking and choosing workers for whom they can alter the model params